src/scripts/active-inactive-orgunit/index.js

```diff
@@ -170,138 +167,158 @@ OrganisationUnitActivator.prototype.processOrgUnitsFromParentLevel = function (d
     console.log("\nPreparing dataValues");
     var _this = this;
     organisationUnits.forEach(function (organisationUnit) {
-        _this.prepareDataSet(organisationUnit, dataElement);
+        _this.prepareDataOrgUnitDataValues(organisationUnit, dataElement, dataValues);
     });
-    console.info("\nPending asyncalls: " + this.asyncCalls);
-    if (this.asyncCalls <= 0) {
-        console.log("Pushing dataValues");
-        this.pushDataValues(this.buildDataValues(this.dataValues), dataElement.dataElementGroup);
-        console.log("The dataValues created from the dataElementGroup " + dataElement.dataElementGroup + " was pushed");
-    }
 };
+
 /**
- * Prepares and pushes the dataSet.
- */
-OrganisationUnitActivator.prototype.prepareDataSet = function (orgUnit, dataElement) {
+ * Prepares the dataSet.*
+@param orgUnit Contains the organisation unit uid and the closed and opening dates used to set the orgUnit as active or inactive
+@param dataElement The dataelement to be pushed with this organisation Unit
+*/
+OrganisationUnitActivator.prototype.prepareDataOrgUnitDataValues = function (orgUnit, dataElement, dataValues) {
     //Parse the server dates.
     if (orgUnit.closedDate != undefined) {
-        orgUnit.closedDate = this.parseDateFromDhis(orgUnit.closedDate);
+        orgUnit.closedDate =  dateFormatter.parseDateFromDhis(orgUnit.closedDate);
     }
     if (orgUnit.openingDate != undefined) {
-        orgUnit.openingDate = this.parseDateFromDhis(orgUnit.openingDate);
+        orgUnit.openingDate = dateFormatter.parseDateFromDhis(orgUnit.openingDate);
     }
 
     var today = new Date();
     today.setMonth(((today.getMonth() + 1) - parseInt(dataElement.periods)));
-    for (var i = parseInt(dataElement.periods); i > 0; i--) {// i > 0 skipes the current period change to 
-        var row;
-        var fixDate = i;
-        var date = new Date();
-        date.setMonth(((date.getMonth()) - fixDate));//get the actual period date
-        var firstPeriodDate = new Date();
-        firstPeriodDate.setMonth(((date.getMonth()) - parseInt(dataElement.periods)));//get the first period date
-        var dateAsPeriod = this.parseDateToPeriodFormat(date);
-        var firstDateAsPeriod = this.parseDateToPeriodFormat(firstPeriodDate);
+    
+    for (var fixDate = parseInt(dataElement.periods); fixDate > 0; fixDate--) {
+        var row; 
+        var date = new Date(); 
+        //Fix the date to show the actual period date month
+        date.setMonth(((date.getMonth()) - fixDate));
+
+        var firstPeriodDate = new Date(); 
+        firstPeriodDate.setMonth(((date.getMonth()) - parseInt(dataElement.periods)));
+
+        var dateAsPeriod = dateFormatter.parseDateToPeriodFormat(date);
+        row = { "dataElement": dataElement.id, "period": dateAsPeriod, "orgUnit": orgUnit.id, "value": "" };
+
+        var firstDateAsPeriod = dateFormatter.parseDateToPeriodFormat(firstPeriodDate);
```

Comment:
Are you sure there is no way to simplify this whole if/else structure and make it simpler? Give it a though please!